components/accelerator/nvidia/query/gpu_memory.go

```diff
@@ -7,29 +7,61 @@ import "strings"
 // ref. https://docs.nvidia.com/deploy/a100-gpu-mem-error-mgmt/index.html#supported-gpus
 func GetMemoryErrorManagementCapabilities(gpuProductName string) MemoryErrorManagementCapabilities {
 	p := strings.ToLower(gpuProductName)
-	switch {
-	case strings.Contains(p, "h100"):
-		return MemoryErrorManagementCapabilities{
+	for _, cap := range supportedMemoryMgmtCaps {
+		if strings.Contains(p, cap.gpuProduct) {
+			return cap.capabilities
+		}
+	}
+	return MemoryErrorManagementCapabilities{}
+}
+
+type supportedMemoryMgmtCap struct {
+	gpuProduct   string
+	capabilities MemoryErrorManagementCapabilities
+}
+
+// ref. https://docs.nvidia.com/deploy/a100-gpu-mem-error-mgmt/index.html#supported-gpus
+var supportedMemoryMgmtCaps = []supportedMemoryMgmtCap{
```

Comment:
add a func called `SupportedMemoryMgmtCapsByGPUProduct`.

In that func, does a longest string match. and return than. i do not want to rely on ordering.

Reply:
done